sys/ztimer64/ztimer64.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2021 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2021 Freie Universität Berlin
+ *               2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_ztimer
+ * @{
+ *
+ * @file
+ * @brief       ztimer64 core functionality
+ *
+ * This file contains ztimer64's main API implementation and functionality
+ *
+ * TODO:
+ * - add some kind of tick so base_clock overflows don't get lost
+ * - TEST
+ * - add proper unittests
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+#include <assert.h>
+#include <stdint.h>
+#include <inttypes.h>
+
+#include "ztimer64.h"
+
+#define ZTIMER64_CHECKPOINT_INTERVAL (1LU << 31)
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static int _add_entry_to_list(ztimer64_clock_t *clock, ztimer64_base_t *entry);
+static int _del_entry_from_list(ztimer64_clock_t *clock,
+                                ztimer64_base_t *entry);
+static void _ztimer64_update(ztimer64_clock_t *clock);
+
+/* debug aid */
+void ztimer64_clock_print(const ztimer64_clock_t *clock);
```

Comment:
static?

Reply:
Is this something you want to be able to insert around?

Reply:
Yup, let's keep it around. :)